webapp/javascript/components/TimelineChart/TimelineTooltip.tsx

```diff
@@ -0,0 +1,195 @@
+      params.xToTime = position.x;
+    }
+
+    plot.hooks.drawOverlay.push(() => {
+      const options = plot.getOptions();
+      const onHoverDisplayTooltip = options?.onHoverDisplayTooltip;
+      const { xaxis } = plot.getAxes() as ShamefulAny;
+      const data = plot.getData();
+
+      if (onHoverDisplayTooltip && exploreTooltip?.length) {
+        const align = params.canvasX > plot.width() / 2 ? 'left' : 'right';
+        const timezone = options.xaxis.timezone;
+        const timeLabel = selection.active
+          ? `${getFormatLabel({
+              date: selection.from,
+              xaxis,
+              timezone,
+            })} - ${getFormatLabel({
+              date: selection.to,
+              xaxis,
+              timezone,
+            })}`
+          : getFormatLabel({
+              date: params.xToTime,
+              xaxis,
+              timezone,
+            });
+
+        const values = data?.map(
+          (
+            d: {
+              data: number[][];
+              tagName: string;
+              color: { color: number[] };
+            },
+            i
+          ) => {
+            let closest = null;
+            let color = null;
+            let tagName = String(i);
+
+            if (d?.data?.length && params.xToTime && params.pageX > 0) {
+              color = d?.color?.color;
+              tagName = d.tagName;
+              closest = (d?.data || []).reduce(function (prev, curr) {
+                return Math.abs(curr?.[0] - params.xToTime) <
+                  Math.abs(prev?.[0] - params.xToTime)
+                  ? curr
+                  : prev;
+              });
+            }
+
+            return {
+              closest,
+              color,
+              tagName,
+            };
+          }
+        );
+
+        const Tooltip: React.ReactElement<
+          ExploreTooltipProps,
+          string | React.JSXElementConstructor<ExploreTooltipProps>
+        >[] = onHoverDisplayTooltip({
+          pageX: params.pageX,
+          pageY: params.pageY,
+          timeLabel,
+          values,
+          align,
+        });
+
+        ReactDOM.render(Tooltip, exploreTooltip?.[0]);
+      }
+    });
+
+    const onMouseUp = () => {
+      selection.active = false;
+    };
+
+    const onMouseDown = () => {
+      selection.active = true;
+    };
+
+    const onPlotSelecting = (
+      e: EventType,
+      selectionData: { xaxis: { from: number; to: number } }
+    ) => {
+      selection.from = selectionData?.xaxis?.from || -1;
+      selection.to = selectionData?.xaxis?.to || -1;
+    };
+
+    plot.hooks.bindEvents.push((p: PlotType, eventHolder: EventHolderType) => {
+      eventHolder.mousemove(onMouseMove);
+      eventHolder.mouseleave(onMouseLeave);
+      eventHolder.mouseup(onMouseUp);
+      eventHolder.mousedown(onMouseDown);
+      plot.getPlaceholder().bind('plothover', onPlotHover);
+      // detect plotselecting event from ./TimelineChartSelection.ts
```

Comment:
not following, why does this tooltip plugin need to know about selection?

Reply:
when we're selecting time range it's showing from-to:
![image](https://user-images.githubusercontent.com/7372044/188167952-165398d2-4f39-467e-bda3-77fad378ef1d.png)

Reply:
honestly i feel like showing that interval is not really useful ATM

Reply:
removed this part